## 예측 정확도 측정하기

In [1]:
from sklearn.datasets import load_iris
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import accuracy_score

In [17]:
iris = load_iris()
dtc = DecisionTreeClassifier()
train_data = iris.data
train_label = iris.target

In [14]:
print(train_data.shape)
print(train_label.shape)

(150, 4)
(150,)


In [18]:
dtc.fit(train_data, train_label)

DecisionTreeClassifier()

In [19]:
# 학습 데이터 세트로 예측 수행
pred = dtc.predict(train_data)
print("예측 정확도:",accuracy_score(train_label,pred))

예측 정확도: 1.0


training 데이터로 accuracy를 측정해서 100%가 나옴

### 학습/테스트 데이터 세트 분리
- `train_test_split()`

In [20]:
from sklearn.model_selection import train_test_split

In [21]:
x_train,x_test,y_train,y_test = train_test_split(iris.data, iris.target, 
                                                test_size = 0.3, random_state=121)

In [22]:
dtc.fit(x_train, y_train)

DecisionTreeClassifier()

In [23]:
pred = dtc.predict(x_test)
print("예측 정확도:",accuracy_score(y_test,pred))

예측 정확도: 0.9555555555555556


## 교차 검증

In [24]:
from sklearn.model_selection import KFold
import numpy as np

In [25]:
features = iris.data
label = iris.target
dtf = DecisionTreeClassifier(random_state=156)

5개의 폴드 세트로 분리하는 `KFold` 객체와 폴드 세트별 정확도를 담을 리스트 객체 생성

In [30]:
kfold = KFold(n_splits=5)
cv_accuracy = []
print("붓꽃 데이터 세트 크기:",features.shape[0])

붓꽃 데이터 세트 크기: 150


150개를 5개의 폴드로 분리했으니 학습 데이터는 120개, 검증 데이터는 30개

In [31]:
n_iter =0

`KFold` 객체의 `split()`은 fold별 학습/검증 테스트의 low index를 array로 반환해준다.

In [ ]:
for train_index